include/node.hpp

```diff
@@ -708,6 +711,31 @@ class node : protected std::tuple<Arguments...> {
         // case sources: HW triggered vs. generating data per invocation (generators via Port::MIN)
         // case sinks: HW triggered vs. fixed-size consumer (may block/never finish for insufficient input data and fixed Port::MIN>0)
 
+        std::size_t n_samples_to_consume = ports_status.in_samples; // default stride == 0
+        if (stride != 0) {
+            const bool first_time_stride = stride_counter == 0;
+            if (first_time_stride) {
```

Comment:
I wonder if the order of when consuming samples/skipping samples by the amount of 'stride' needs to be inverted. :thinking: 

I.e. when starting the processing we'd like to first process the initial N samples and only then to start skipping samples by 'stride'. Could you check? 

Also, please maybe add some more small bits of code documentation since this logic is important and for the uninitiated hard-to-read on the first go. Thanks in advance.

Reply:
I think the implementation is correct. If `first_time_stride` the sample processing is done as usual. I added some comments in the code. Please review this part once again. I will also add some QA tests which check also sequence correctness.

Reply:
I like the additional code documentation. :+1: